src/test/end.t.sol

```diff
@@ -483,6 +504,70 @@ contract EndTest is DSTest {
         assertEq(balanceOf("gold", address(gold.gemA)), 0);
     }
 
+    // -- Scenario where there is one collateralised CDP
+    // -- undergoing auction at the time of cage
+    function test_cage_snip() public {
+        Ilk memory gold = init_collateral("gold");
+
+        Usr ali = new Usr(vat, end);
+
+        vat.fold("gold", address(vow), int256(ray(0.25 ether)));
+
+        // Make a CDP:
+        address urn1 = address(ali);
+        gold.gemA.join(urn1, 10 ether);
+        ali.frob("gold", urn1, urn1, urn1, 10 ether, 15 ether);
+        (uint ink1, uint art1) = vat.urns("gold", urn1); // CDP before liquidation
+        (, uint rate,,,) = vat.ilks("gold");
+
+        assertEq(vat.gem("gold", urn1), 0);
+        assertEq(rate, ray(1.25 ether));
+        assertEq(ink1, 10 ether);
+        assertEq(art1, 15 ether);
+
+        vat.file("gold", "spot", ray(1 ether)); // Now unsafe
+
+        uint id = dog.bark("gold", urn1);                   
+        (, uint256 tab, uint256 lot,,,) = gold.clip.sales(id);  
+        assertEq(tab, art1 * rate * 1.1 ether / WAD); // tab uses chop             
+        assertEq(dog.Dirt(), tab);
+        assertEq(lot, ink1);
+
+        {
+            (uint ink2, uint art2) = vat.urns("gold", urn1); // CDP after liquidation
+            assertEq(ink2, 0);
+            assertEq(art2, 0);
+        }
+
+        // Collateral price is $5
+        gold.pip.poke(bytes32(5 * WAD));
+        spot.poke("gold");
+        end.cage();
+        end.cage("gold");
+        assertEq(end.tag("gold"), ray(0.2 ether)); // par / price = collateral per DAI
+
+        assertEq(vat.gem("gold", address(gold.clip)), lot); // From grab in dog.bark()
+        assertEq(vat.sin(address(vow)),       art1 * rate); // From grab in dog.bark()
+        assertEq(vat.vice(),                  art1 * rate); // From grab in dog.bark()
+        assertEq(vat.debt(),                  art1 * rate); // From frob
+        assertEq(vat.dai(address(vow)),                 0); // vat.suck() hasn't been called 
+
+        end.snip("gold", id);
+
+        assertEq(dog.Dirt(),                            0); // From clip.yank()
```

Comment:
I'd add assertions on the Clipper state as well--`sales` entry cleared, and `id` removed from the list of active auctions. Otherwise, we have no test coverage for the state changes that `yank` is supposed to effect.